README.md

```diff
@@ -40,27 +40,25 @@ It will use the highest version that satisfies the range.
 ## Automatic switching
 Manual switching can be avoided.  Add the following below to your environment depending on your shell choice.
 
-### Zsh
+### Zsh and Bash
 
-```bash
-echo "\nchpwd () {\n nodengine\n}" >> ~/.zshrc
-```
-### Bash
-
-```bash
+```shell
 # Enable nodengine autoswitching
-cd () { builtin cd "$@" && chpwd; }
-pushd () { builtin pushd "$@" && chpwd; }
-popd () { builtin popd "$@" && chpwd; }
-chpwd () {
-  FILE=$PWD/package.json
-
-  if [ -f $FILE ];
-  then
-     nodengine
-  fi
-}
-```  
+if hash nodengine 2>/dev/null; then
```

Comment:
Hey @Kikobeats - sorry for the late reply. I know you already closed the PR, but figured I'd respond anyways.

The `which` command is not consistent across different environments, and should be avoided – some systems don't even have return codes, some use PATH info from `~/.csh` which may or not be used on the system, and it's also an external command rather than a built-in.

The `hash` or `command` built-in's are more consistent, accurate, and faster.